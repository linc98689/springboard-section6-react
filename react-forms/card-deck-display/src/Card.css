.Card{
    /* position:fixed;
    background-color: blue;
    width: 50x;
    height: 100px; 
    top: 10%;
    left: 0%;
     transform: translate(0%, -50%); */

}

.Card-img{
    position:absolute;
    /* display: block; */
    left:80px;
    top:150px;
    /* transform: translate(-50%, 0); */
    /* transform: translate(-50%, -50%); */
    width: 100px;
    /* height: 200px; */
}