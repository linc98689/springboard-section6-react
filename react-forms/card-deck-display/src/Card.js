import React from "react";
import "./Card.css";

const Card = ({src, id})=>{
    let angle = id * 25;
    return (
        <div className="Card"  >
            <img className="Card-img" src={`${src}`} alt="card"
            style={{transform:`rotate(${angle}deg)`}}/>
        </div>
    );
};

export default Card;