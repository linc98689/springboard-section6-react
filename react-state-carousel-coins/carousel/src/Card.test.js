import { render, fireEvent } from "@testing-library/react";
import Card from "./Card";

// data for testing
const testProp = {
    caption:"Test Photo",
    src:"test.com",
    curNum:1,
    totalNum:3
};

// Smoke test
it("should render", ()=>{
    render(<Card caption={testProp.caption} src={testProp.src} curNum={testProp.curNum} totalNum={testProp.totalNum}/>);
}); 

// snapshot test
it("should not change", ()=>{
    const {asFragment} = render(
        <Card caption={testProp.caption} src={testProp.src} curNum={testProp.curNum} totalNum={testProp.totalNum}/>
    );

    expect(asFragment()).toMatchSnapshot();
});

