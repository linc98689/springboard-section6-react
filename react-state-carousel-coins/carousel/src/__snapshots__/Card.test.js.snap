// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should change 1`] = `
<DocumentFragment>
  <div
    class="Card"
  >
    <h4
      class="Card-title"
    >
      Test Photo
    </h4>
    <img
      alt="Test Photo"
      class="Card-image"
      src="test.com"
    />
    <small
      class="Card-small"
    >
      Image  of 3.
    </small>
  </div>
</DocumentFragment>
`;

exports[`should not change 1`] = `
<DocumentFragment>
  <div
    class="Card"
  >
    <h4
      class="Card-title"
    >
      Test Photo
    </h4>
    <img
      alt="Test Photo"
      class="Card-image"
      src="test.com"
    />
    <small
      class="Card-small"
    >
      Image  of 3.
    </small>
  </div>
</DocumentFragment>
`;
